src/org/traccar/protocol/FlespiProtocolDecoder.java

```diff
@@ -0,0 +1,111 @@
+import org.jboss.netty.handler.codec.http.HttpResponse;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.jboss.netty.handler.codec.http.HttpVersion;
+import org.jboss.netty.handler.codec.http.HttpHeaders;
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.DeviceSession;
+import org.traccar.model.Position;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonNumber;
+import javax.json.JsonObject;
+import java.io.StringReader;
+import java.net.SocketAddress;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.LinkedList;
+import java.util.Date;
+
+public class FlespiProtocolDecoder extends BaseProtocolDecoder {
+
+    private final String protocolName;
+
+    public FlespiProtocolDecoder(FlespiProtocol protocol) {
+        super(protocol);
+        protocolName = protocol.getName();
+    }
+
+    @Override
+    protected Object decode(
+            Channel channel, SocketAddress remoteAddress, Object msg) throws Exception {
+
+        HttpRequest request = (HttpRequest) msg;
+        JsonArray result = Json.createReader(new StringReader(request.getContent().toString(StandardCharsets.UTF_8)))
+                .readArray();
+        List<Position> positions = new LinkedList<>();
+        for (int i = 0; i < result.size(); i++) {
+            JsonObject message = result.getJsonObject(i);
+            String ident = message.getString("ident");
+            if (ident == null) {
+                continue;
+            }
+            DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, ident);
+            if (deviceSession == null) {
+                continue;
+            }
+            Position position = new Position();
+            position.setDeviceId(deviceSession.getDeviceId());
+            decodePosition(message, position);
+            positions.add(position);
+        }
+
+        sendResponse(channel, HttpResponseStatus.OK);
+        return positions;
+    }
+
+    private void sendResponse(Channel channel, HttpResponseStatus status) {
+        if (channel != null) {
+            HttpResponse response = new DefaultHttpResponse(HttpVersion.HTTP_1_1, status);
+            response.headers().add(HttpHeaders.Names.CONTENT_LENGTH, 0);
+            channel.write(response);
+        }
+    }
+
+    private void decodePosition(JsonObject object, Position position) {
+        position.setProtocol(protocolName);
+
+        position.setTime(new Date((long) object.getJsonNumber("timestamp").doubleValue() * 1000));
+        JsonNumber lat = object.getJsonNumber("position.latitude");
+        JsonNumber lon = object.getJsonNumber("position.longitude");
+        position.setLatitude(lat != null ? lat.doubleValue() : 0);
+        position.setLongitude(lon != null ? lon.doubleValue() : 0);
+
+        JsonNumber speed = object.getJsonNumber("position.speed");
+        position.setSpeed(speed != null ? speed.doubleValue() : 0);
```

Comment:
I would recommend to change this and other places to `if` code blocks like this:
```java
if (speed != null) {
    position.setSpeed(speed.doubleValue());
}
```

It's easier to read and no need for fallback zero value.